hack/third-party/capx/go.mod

```diff
@@ -3,67 +3,41 @@
 
 module github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/external/capx
 
-go 1.21
+go 1.22.1
```

Comment:
nit: Please keep as `1.22`
```suggestion
go 1.22
```

Reply:
I think you can fix here separately though right?

Reply:
sure. was trying to avoid  make apis.sync to overwrite it again. sure we can do it separately

Reply:
seems like pre-commit overwrites it so we need to fix it in capx only first

Reply:
Oh interesting because it's declared as such in CAPX it's the minimum this module can use too... Still learning about these go module intricacies 😅

Reply:
Looks like we do need to fix it upstream first :( 
The tests are failing

Reply:
🤔 can't we just add 1.22.1 in this module for now? Shouldn't affect anything else once the types are generated. Revert my suggestion basically 😅

Reply:
https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/410/files